KFSPTS-17250 update pom version to the 8/8/2019 version of financials

db483 opened the pull request.

cah292 commented on src/main/java/org/kuali/kfs/fp/document/document/validation/DisbursementVoucherEmployeeInformationValidation.java:
```diff
@@ -0,0 +1,104 @@
+
+    protected AccountingDocument accountingDocumentForValidation;
+    protected DataDictionaryService dataDictionaryService;
+    protected PersonService personService;
```
It looks like this overlay was added just to increase the visibility of the service instance variables. Would it be helpful to add some comments or other documentation to make the intent of this overlay clearer, as well as whether it's pulling in a future KualiCo fix?

db483 replied:
Good point. I will add a comment and I have reported this to KualiCo.

cah292 commented on src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/UserProcurementProfile.xml:
```diff
@@ -149,19 +162,23 @@
 
-  <dd:boAttributeRef id="UserProcurementProfile-profileUser.principalName-parentBean" parent="PersonImpl-principalName" abstract="true" 
-  attributeName="profileUser.principalName" label="User Principal Name" shortLabel="Principal Name" required="false">
-    <dd:user universalIdAttribute="principalId" userObjectAttribute="profileUser"/>
```
This "dd:user" tag doesn't seem to have been translated into the equivalent Spring XML by this rewrite. Can you please add the related Spring XML for this if necessary? I believe this would translate to a "KualiUserControlDefinition" bean.

cah292 commented on src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/UserProcurementProfile.xml:
```diff
@@ -149,19 +162,23 @@
 
-  <dd:boAttributeRef id="UserProcurementProfile-profileUser.name-parentBean" parent="PersonImpl-name" abstract="true" 
-  attributeName="profileUser.name" label="Person Name" shortLabel="Person Name" required="false">
-    <dd:user universalIdAttribute="principalId" userObjectAttribute="profileUser"/>
```
Another "dd:user" tag that needs to be translated.

cah292 commented:
There are a few other items that could use cleanup in this PR or in future PRs:

[1] At least the following classes could have their SpringContext.getBean() calls cleaned up, but I'm okay deferring them to a future user story if the code runs fine with them as-is:

* ProcurementCardCreateDocumentServiceImpl
* CuDisbursementVoucherAccountingLineTotalsValidation
* CuDisbursementVoucherPayeeInitiatorValidation

[2] There are a few files that may still need to have the "dd" XML namespace information removed from them:

* AccountMaintenanceDocument.xml (for the CG module)
* SubAccountMaintenanceDocument.xml

cah292 commented on src/main/resources/edu/cornell/kfs/fp/document/validation/configuration/FinancialProcessingValidators.xml:
```diff
@@ -25,7 +25,7 @@
 	<bean id="DisbursementVoucher-employeeInformationValidation" class="edu.cornell.kfs.fp.document.validation.impl.CuDisbursementVoucherEmployeeInformationValidation" abstract="true" />
```
This bean now needs to have its services injected. Some of the other beans in this file may need to be updated likewise.

db483 replied:
Added service injection for beans in FinancialProcessingValidation.xml

db483 commented on src/main/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherExtractionHelperServiceImpl.java:
```diff
@@ -32,6 +32,7 @@
 import org.kuali.kfs.sys.KFSConstants;
 import org.kuali.kfs.sys.businessobject.SourceAccountingLine;
 import org.kuali.kfs.sys.context.SpringContext;
```
Good point. Would it make more sense to create a separate user story?

db483 commented on src/main/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherExtractionHelperServiceImpl.java:
```diff
@@ -454,4 +455,8 @@ private boolean isRecurringDV(DisbursementVoucherDocument paymentSource) {
 	       return isRecurringDV;
 	   }
+	   
+	   public PaymentSourceHelperService getPaymentSourceHelperService() {
```
This is still used in RecurringDisbursementVoucherDocumentServiceImpl.

cah292 replied:
Ok, thanks for explaining. I also see that we have a KFSPTS-6498 user story for cleaning up the recurring DV service's usage of one of the PaymentSourceHelperService methods, which should allow us to eventually remove this method.

db483 commented on src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherEmployeeInformationValidation.java:
```diff
@@ -43,7 +43,7 @@ public boolean validate(AttributedDocumentEvent event) {
         
         String employeeId = payeeDetail.getDisbVchrPayeeIdNumber();
-        Person employee = SpringContext.getBean(PersonService.class).getPersonByEmployeeId(employeeId);
+        Person employee = personService.getPersonByEmployeeId(employeeId);
```
I cleaned up the remaining instances

db483 commented:
[1] 

ProcurementCardCreateDocumentServiceImpl
* cleaned up the ones from base financials; new jira for remaining
CuDisbursementVoucherAccountingLineTotalsValidation
* new jira
CuDisbursementVoucherPayeeInitiatorValidation
* base financials service variables are private; will open ticket with Kualico to increase visibility

[2] 

AccountMaintenanceDocument.xml (for the CG module)
* updated
SubAccountMaintenanceDocument.xml
* updated

cah292 commented on src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/UserProcurementProfile.xml:
```diff
@@ -149,19 +162,31 @@
+        p:name="profileUser.principalName" p:label="User Principal Name"
+        p:shortLabel="Principal Name" p:required="false">
+      <property name="control">
+          <bean parent="KualiUserControlDefinition" p:personNameAttributeName="profileUser.principalName"
```
Should the personNameAttributeName be referencing "profileUser.name" instead?

cah292 commented on src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/UserProcurementProfile.xml:
```diff
@@ -149,19 +162,31 @@
+      <property name="control">
+          <bean parent="KualiUserControlDefinition" p:personNameAttributeName="profileUser.name"
+                p:universalIdAttributeName="principalId"
+                p:userIdAttributeName="profileUser.name"/>
```
Should the userIdAttributeName be referencing "profileUser.principalName" instead?
